go/libraries/doltcore/schema/schema_impl.go

```diff
@@ -44,7 +44,7 @@ func SchemaFromCols(allCols *ColCollection) Schema {
 	}
 
 	if len(pkCols) == 0 {
-		panic("no primary key columns specified")
```

Comment:
should this be legal?

Reply:
Maybe we should enforce pks on schemas in a different place.  Such as when you attempt to create a table with the schema.

Reply:
I think we must enforce it in other places already, otherwise we would be seeing a bunch of panics.

Reply:
We have UnkeyedSchemaFromColsfor this, which doesn't enforce the primary key check. We use that for e.g. SQL result sets and other unkeyed schemas.